zcash_client_backend/src/data_api.rs

```diff
@@ -472,22 +482,31 @@ impl SentTransactionOutput {
 pub struct AccountBirthday {
     height: BlockHeight,
     sapling_frontier: Frontier<Node, NOTE_COMMITMENT_TREE_DEPTH>,
+    recover_until: Option<BlockHeight>,
 }
 
 impl AccountBirthday {
     /// Constructs a new [`AccountBirthday`] from its constituent parts.
     ///
     /// * `height`: The birthday height of the account. This is defined as the height of the last
-    /// block that is known to contain no transactions sent to addresses belonging to the account.
+    ///    block that is known to contain no transactions sent to addresses belonging to the account.
     /// * `sapling_frontier`: The Sapling note commitment tree frontier as of the end of the block
-    /// at `height`.
+    ///    at `height`.
+    /// * `recover_until`: An optional height at which the wallet should exit "recovery mode". In
+    ///    order to avoid confusing shifts in wallet balance and spendability that may temporarily be
+    ///    visible to a user during the process of recovering from seed, wallets may optionally set a
+    ///    "recover until" height. The wallet is considered to be in "recovery mode" until there
+    ///    exist no unscanned ranges between the wallet's birthday height and the provided
+    ///    `recover_until` height, exclusive.
     pub fn from_parts(
         height: BlockHeight,
         sapling_frontier: Frontier<Node, NOTE_COMMITMENT_TREE_DEPTH>,
+        recover_until: Option<BlockHeight>,
```

Comment:
This will be used in #914 to provide a "recovery status" endpoint in addition to (or as part of) the "sync status" support.